### ISSUE TITLE
Error on some websites "undefined is not an object (evaluating 'a[Symbol.iterator]')"

### ISSUE THREAD

**Issue by kamilms21** [gildas-lormeau/single-file-cli#124]
I am unable to take a screenshot on certain websites, such as YouTube.com, due to the following error:

I am using a Docker container with Playwright (it fails on the singlefile.getPageData method), but I can also see that taking screenshots doesn’t work when using the CLI.

An example of a page where the error "undefined is not an object (evaluating 'a[Symbol.iterator]')" occurs is: https://www.youtube.com/watch?v=RR5Y_b9VajM&list=RDRR5Y_b9VajM&start_radio=1

**Comment by gildas-lormeau** [comment]
I was not able to reproduce the issue, what browser/os and versions do you use?

**Comment by kamilms21** [comment]
I am running a script in a Docker container with Playwright, and the same issue appears whether I try Chromium, WebKit, or Firefox.

I created a repository where I reproduced this issue: https://github.com/kamilms21/single-file-test

You just need to install the dependencies and run "npx playwright test --project chromium."

It seems to me that you couldn't reproduce the issue because it only appears if the video player is visible, and depending on the environment, YouTube shows a cookie modal first, and the rest of the page won’t load until the cookies are accepted. I think I saw a similar problem on another site besides YouTube, but I can't remember which one.

Thanks for looking into this

<img width="1099" alt="Screenshot 2024-09-22 at 13 08 41" src="https://github.com/user-attachments/assets/b734a62e-8cff-4b73-98fc-407f4d838916">

**Comment by gildas-lormeau** [comment]
Thank you, I was able to reproduce the issue. This is due to a bug in polymer and/or (?) playwright. The [`Element#replaceWith()`](https://developer.mozilla.org/en-US/docs/Web/API/Element/replaceWith) implementation is not compliant to the spec and triggers an unexpected error in some cases. I added a try/catch statement to handle it. I also updated the built code (e.g. `single-file-bundle.js`) in the repository.

I hope this is the only Playwright bug, as it's one of the reasons I removed it from `single-file-cli` and opted for a CDP implementation.
